src/matlab2tikz.m

```diff
@@ -2600,6 +2602,18 @@ function assertRegularAxes(handle)
             plotOptions = opts_add(plotOptions,'contour/labels','false');
         end
 
+        % Get line properties
+        [m2t, lineOptions] = getLineOptions(m2t, h);
```

Comment:
Actually ignore my previous comment that I deleted, it is this line that adds unconditionally the `solid`:

```
[~,lineOpt] = getLineOptions(m2t,h)
lineOpt = 
    'solid'    ''
```

We should probably test for default here too?